itest/lnd_channel_backup_test.go

```diff
@@ -1672,3 +1954,383 @@ func assertDLPExecuted(ht *lntest.HarnessTest,
+		// defaultCSV-1 and we already mined one block after the
+		// commitmment was published, so take that into account.
+		ht.MineEmptyBlocks(int(defaultCSV - blocksMined))
+
+		// Carol should have two sweep requests - one for her commit
+		// output and the other for her anchor.
+		ht.AssertNumPendingSweeps(carol, 2)
+
+		// Mine a block to trigger the sweep in Carol.
+		ht.MineEmptyBlocks(1)
+		ht.MineBlocksAndAssertNumTxes(1, 1)
+
+		// Now the channel should be fully closed also from
+		// Carol's POV.
+		ht.AssertNumPendingForceClose(carol, 0)
+
+		// We'll now mine the remaining blocks to prompt Dave to sweep
+		// his CLTV-constrained output.
+		resp := dave.RPC.PendingChannels()
+		blocksTilMaturity :=
+			resp.PendingForceClosingChannels[0].BlocksTilMaturity
+		require.Positive(ht, blocksTilMaturity)
+
+		ht.MineEmptyBlocks(int(blocksTilMaturity))
+
+		// Dave should have two sweep requests - one for his commit
+		// output and the other for his anchor.
+		ht.AssertNumPendingSweeps(dave, 2)
+
+		// Mine a block to trigger the sweep.
+		ht.MineEmptyBlocks(1)
+		ht.MineBlocksAndAssertNumTxes(1, 1)
+
+		// Now Dave should consider the channel fully closed.
+		ht.AssertNumPendingForceClose(dave, 0)
+	} else {
+		// Carol should sweep her funds immediately, as they are not
+		// timelocked. We also expect Carol and Dave sweep their anchors
+		// if it's an anchor channel. Dave's funds are timelocked, so
+		// expect just one sweep.
+		if lntest.CommitTypeHasAnchors(commitType) {
+			ht.AssertNumPendingSweeps(carol, 2)
+		} else {
+			ht.AssertNumPendingSweeps(carol, 1)
+		}
+		ht.AssertNumPendingSweeps(dave, 1)
+
+		// Mine one block to trigger the sweeper to sweep.
+		ht.MineEmptyBlocks(1)
+		blocksMined++
+
+		// Expect one tx - the commitment sweep from Carol. For
+		// anchor channels, we expect the two anchor sweeping
+		// txns to be failed due they are uneconomical.
+		ht.MineBlocksAndAssertNumTxes(1, 1)
+		blocksMined++
+
+		// Now Carol should consider the channel fully closed.
+		ht.AssertNumPendingForceClose(carol, 0)
+
+		// After Dave's output matures, he should also reclaim his
+		// funds.
+		//
+		// The commit sweep resolver publishes the sweep tx at
+		// defaultCSV-1 and we already have blocks mined after the
+		// commitmment was published, so take that into account.
+		ht.MineEmptyBlocks(int(defaultCSV - blocksMined))
+
+		// Mine one block to trigger the sweeper to sweep.
+		ht.MineEmptyBlocks(1)
+
+		// Dave's sweeper swept the outputs of force close tx.
+		if lntest.CommitTypeHasAnchors(commitType) {
+			ht.AssertNumPendingSweeps(dave, 2)
+		} else {
+			ht.AssertNumPendingSweeps(dave, 1)
+		}
+
+		// Assert the sweeping tx is mined.
+		ht.MineBlocksAndAssertNumTxes(1, 1)
+
+		// Funds are now in Dave's wallet.
+		ht.AssertNumPendingForceClose(dave, 0)
+
+		// Now the channel should be fully closed also from
+		// Carol's POV.
+		ht.AssertNumPendingForceClose(carol, 0)
```

Comment:
Same here, I think we can reduce the test quite a bit, because if something changes for example with the anchor sweep all those tests would need to be changed, so I think we should not repeat tests wdyt ?

Reply:
Fixed